clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/CopybookDownloaderForE4E.ts

```diff
@@ -250,29 +256,63 @@ export class CopybookDownloaderForE4E {
     return false;
   }
 
-  private static getCopybookPath(
+  private static async getCopybookPath(
     instance: string,
     mapped: string,
     downloadFolder: string,
     copybook: string,
-  ): string {
+  ): Promise<string> {
     let folder = CopybookURI.createDatasetPath(
       instance,
       mapped,
       downloadFolder,
       E4E_FOLDER,
     );
 
-    if (!fs.existsSync(folder)) fs.mkdirSync(folder, { recursive: true });
+    /**
+     * There is an issue with VSCode File Watcher on Linux where it
+     * fails to watch subfolders changes when more subfolders are created
+     * all at once.
+     * https://github.com/microsoft/vscode/issues/142694
+     *
+     * As a workaround, the path is splitted into individual subfolders
+     * are they are created incrementally one by one.
+     */
+    const subfoldersPath = folder.replace(`${downloadFolder}${path.sep}`, "");
```

Comment:
maybe introduce a function that will only concatenate the suffix of the path (unlike createDatasetPath)?

Reply:
Even better - write a function that just returns the `subfolders` directly.